### ISSUE TITLE
Midi QoL + Dice So Nice: Auto Check Saves - Result Card Not Showing Up Until Refresh

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#263]
In GitLab by @leetsolutions on Mar 4, 2021, 11:09

When using both of these modules at the same time, there seems to be an issue with the results card showing up. If I use the following settings:

1) Auto Check Saves - Off: 
Works as intended

2) Auto Check Saves - All See Result: 
Dice results are shown in the console, but the result card is not shown unless the page is refreshed.

3) Auto Check Saves - Only GM Sees:
Same as above, except after a refresh the result card is whispered as expected.

4) Auto Check Saves - All See Result + Rolls
The dice are thrown and the rolls are showed in the chat log, but the result card is not shown until a refresh.

I'm not sure if this is an issue in DSN or Midi QoL, so I apologize if I'm reporting something that isn't on your end. The main issue on my end is that I'd ideally like only the GM to be able to see the result or the rolls+result.

**Comment by tposney** [comment]
In GitLab by @fenthik on Mar 4, 2021, 20:10

After checking the issue on my installation more in depth, I can confirm that I experience the same problem.

**Comment by tposney** [comment]
I'm not seeing this behaviour. Can you confirm a few things, are these player or npc saving throws? Are you using better rolls? Are you using the merge card?
Whey you say result card, do you mean the card that reports who saved, or the actual saving throw plus dice formula card.

**Comment by tposney** [comment]
In GitLab by @leetsolutions on Mar 5, 2021, 09:57

The behavior happens with both NPC and Player saving throws. I'm not using Better Rolls, and did my testing with only these mods active:

```
Midi QoL v0.3.75
Dice So Nice v3.3.1
libWrapper v1.3.5.0
Find the culprit v1.3.3
```

The results card that I'm talking about is the one that reports who saved. 

![FoundryVTT_pOXAhE0VQD](https://my-gitlab-bucket.s3.amazonaws.com/640711674/a8e57508aa15b070ad2d06d6c8a914f6f8c71a5d1fb0a4e8b37b6fd9f334d21b/FoundryVTT_pOXAhE0VQD.png)

I'm not using the merge card (but I just tested out of curiosity and it DOES seem to work properly when using the merge card).

Here's a small recording of what I'm talking about. When I initially cast the spell you can see that the rolls happen in the console but the result doesn't show up. But when I refresh the page, the results do show up after.

![zPHzj8m28r](https://my-gitlab-bucket.s3.amazonaws.com/640711674/c4dffacd09ea9620296a41c5f8b091e665dae68fbeaeed8e33835023179365a9/zPHzj8m28r.gif)

**Comment by tposney** [comment]
Thanks finally worked out what was going on. The problem was only occuring if there was no player saving and monster rolls were not being displayed. A fix will go out in 0.3.76

**Comment by tposney** [comment]
In GitLab by @leetsolutions on Mar 7, 2021, 07:56

Sorry to be a bother, but I downloaded v0.3.76 and with the same plugin loadout I'm still experiencing the issue the same as before, whether it's all players, all monsters, or a mix of both saving.

**Comment by tposney** [comment]
Hmm, not sure what is going on. If possible can I connect to you world to see what is happening

**Comment by tposney** [comment]
In GitLab by @leetsolutions on Mar 9, 2021, 03:17

I'm not sure what happened exactly, but I updated do 0.3.80 this morning before seeing your response here, and then started to set up a test world for you to connect to. I tested things again just to make sure I wasn't going to waste your time by having you connect, and to my surprise the issue seemed to be gone!

I'm definitely not crazy because I remember double checking that I was on 0.3.76 before reporting here and I definitely remember having to swap to the merged card when showing my GM how things worked. But everything seems to be working fine in all of my worlds now.

In any case, thanks so much for the fix! Sorry to waste your time with the follow up report.
